tests/test_observable/test_starmap.py

```diff
@@ -405,6 +405,211 @@ def mapper(x, y):
         assert xs.subscriptions == [subscribe(200, 290)]
         assert invoked[0] == 3
 
+    def test_starmap_with_index_throws(self):
```

Comment:
Tests derived from https://github.com/ReactiveX/RxPY/blob/8263b1e36231aeec956e825a991f3350ee3e74a7/tests/test_observable/test_map.py#L265 and adapted for starmap